### ISSUE TITLE
crash with 857d42e

### ISSUE THREAD

**Issue by marilmanen** [eteran/nedit-ng#337]
Following macro seems to cause a crash if the tab from which the macro has been started is closed and a new file from the list dialog is selected for opening. Maybe closing a tab should be prevented while macro command is running?

```
if ($selection_start > -1) {
    string = get_selection()
} else {
    string = string_dialog("Give string to grep")
}
matches = shell_command("grep -ni '" string "' * |& grep -v '^grep: '", "")
if ( matches == "" ) {
    dialog("No matches found for " string)
}
selection = 1
while ( selection != "" && matches != "" ) {
  selection = list_dialog( "Select one of the matching lines___________________________________________________________________________________________________________________________________________________________________________________________________________", matches )
  if ( selection != "" ) {
    file = selection
    file = replace_in_string(file, ":.*", "", "regex")
    line = selection
    line = replace_in_string(line, ":\\D.*", "", "regex")
    line = replace_in_string(line, ".*:", "", "regex")
    if ( ( file == "") || ( line == "" ) ) {
      dialog("Something wrong with this macro: " selection "--" file "__" line, "OK")
    } else {
      open($file_path file)
      focus_window($file_path file)
      goto_line_number(line)
    }
  }
  matches = shell_command("grep -ni '" string "' * |& grep -v '^grep: '", "")
}

```

**Comment by eteran** [comment]
It's been a LONG time since the last crash! (which is great news). I'll definitely see if I can fix this crash ASAP for ya and probably call the result a new release since it's been so long!

**Comment by eteran** [comment]
OK, this one is a bit tricky but I'm looking into it. We actually had a similar issue with a shell command causing a crash if the document was closed out from under it... but that was easier to fix because I could just wait on the process.

There actually is code in there that **attempts** to deal with the macro circumstance, it just isn't quite right. I'll keep you posted.

**Comment by marilmanen** [comment]
It looks I got a false pop-up message with "Error excuting macro: list_dialog can't be called from non-suspendable context". Might be my mistake, but I could not figure out the reason. I'll create a new ticket if I find a way to confirm that it is an issue and not a user mistake.

**Comment by eteran** [comment]
@marilmanen Well, that's also an interesting one for sure. I can probably improve the code but I basically just made it choose not to resume the macro if the document object got deleted.
